### PR TITLE
chore(ci): update github action versions

### PR DESCRIPTION
**Pull request by robinsmith-source**
## 📝 Description

This pull request updates the github action versions to support Node 20, since the Node.js 16 actions are deprecated.

## ⛳️ Current behavior (updates)

Deprecated actions are used.

## 🚀 New behavior

New actions should work perfectly fine, no warning message gets displayed. 

## 💣 Is this a breaking change (Yes/No):

No

## 📝 Additional Information

https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

## Summary by CodeRabbit

- **Documentation**
  - Updated contribution guidelines to require Node.js 20.x and pnpm 9.6.0 or higher.
- **Bug Fixes**
  - Corrected a typo in the bug report template for browser options.
- **Chores**
  - Updated GitHub Actions and workflows to use the latest versions of actions for dependency installation and repository checkout.
- **New Features**
  - Updated the required Node.js and PNPM versions in the package.json to support newer technologies.

### CODE REVIEW

File: .github/common-actions/install/action.yml [hunk 1]
```diff
@@ -5,12 +5,12 @@ runs:
   using: composite
   steps:
     - name: Install dependencies
-      uses: pnpm/action-setup@v2.2.4
+      uses: pnpm/action-setup@v3
```
Comment (ryo-manba) [inline .github/common-actions/install/action.yml:8]: memo:
https://github.com/pnpm/action-setup/releases/tag/v3.0.0

File: .github/workflows/QA.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout branch
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
```
Comment (ryo-manba) [inline .github/workflows/QA.yaml:14]: memo: 
https://github.com/actions/setup-node/releases/tag/v4.0.0

**Review by ryo-manba — APPROVED** [review]
Thanks!

**Comment by robinsmith-source** [comment]
I've updated pnpm and Node.js to their latest versions. If there are any incompatibilities, I can revert those changes and only upgrade the gh-action versions here.

File: .changeset/long-boxes-joke.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+---
+---
```
Comment (wingkwong) [inline .changeset/long-boxes-joke.md:2]: we add changeset for packages only. please delete this file since your changes don't require one.

Comment (robinsmith-source) [reply .changeset/long-boxes-joke.md:2]: I was unsure whether to do so, but in your Contribution.md it's described as:

> "If you made minor changes like CI config, prettier, etc., you can run pnpm changeset add --empty to generate an empty changeset file to document your changes."

Overall, this changeset shouldn't affect versioning at all.

**Pull request merged**